### ISSUE TITLE
Have some place to share the scan files

### ISSUE THREAD

**Issue by klieret** [celis/B_decays_clustering#10]
It would probably be a good idea to set up some place to share the outputs of ``scan.py`` that take a long time to compute.

For now I just uploaded mine (np grid: 10, q2 bins: 10) here:

https://files.fm/u/9zyyyhx8 

I'm currently running scan.py with np grid 30, q2 bins 10 so that should be available in around 24h (server is a bit crowded).

**Comment by celis** [comment]
yes thats true,

**Comment by klieret** [comment]
Also I just noticed that for bigger grids, RAM becomes a huge issue. When doing the hierarchy.linkage on a 30x30x30 grid, it consumes up to 5 or 6 GB and everything will freeze if you don't have that available. 

Anyway, here's a couple of results (because of the RAM issue I've directly uploaded the clustering results for d=0.1, 0.75 and 0.5, corresponding to 3, 4 and 6 clusters):

https://files.fm/u/chpypfa6

**Comment by klieret** [comment]
Here's what the notebook looks like with that data:

https://files.fm/u/kfeb7z7n 

But anyone should get these same results with the data file from above.

Note that instead of doing ``plot_cluster(df, cols)``, I changed the implementation and one now does ``cp = ClusterPlot(df)`` and then ``df.scatter(cols)`` or ``df.fill(cols)`` (for new color plots).

![image](https://user-images.githubusercontent.com/13602468/47836643-71703700-ddec-11e8-8683-c513e0b0edb8.png)

**Comment by celis** [comment]
Hi 

"Also I just noticed that for bigger grids, RAM becomes a huge issue. When doing the hierarchy.linkage on a 30x30x30 grid, it consumes up to 5 or 6 GB and everything will freeze if you don't have that available."

I understand the hierarchical clustering is among the most expensive clustering we can do, as it keeps all the structure.   If we use other clustering algorithms we will likely not have this problem so severe.

 Another advantage of trying another clustering algorithm is that they can produce centers for each cluster and thats very useful for us.  I will try to play with some of the clustering methods in scikit-learn.  

Another thing I will try is to train some classifier like random forest with the output of the clustering algorithm, then it can classify any point of the parameter space.

**Comment by celis** [comment]
For the grid, I think a 30x30x30 grid is already quite good, dont think there is need to go beyond that.   For the q2 grid we can use 15 bins as a upper limit I think (based on the experimental results published so far).

**Comment by klieret** [comment]
Yes, playing around with other clustering algorithms sounds great!

Not completely sure about the machine learning idea. Because as you said, 30^3 is already quite fine of a grid and if needed we could just make that even finer. So in my opinion, a deterministic approach to assign a new point that is between the grid lines to an existing cluster would be more transparent and probably sufficiently precise (plus easier from the implementation I would say).

**Comment by klieret** [comment]
I will add a method switch to the cluster method so that we can have multiple methods in parallel.

**Comment by klieret** [comment]
Since e.g. not using Hierarchical clustering would change the implementation quite a lot, it's probably best to have subclasses of the ``Cluster`` class, i.e. one subclass ``HierarchicalCluster`` and one ``KmeansCluster`` for k-means clustering etc.

**Comment by celis** [comment]
"Not completely sure about the machine learning idea. Because as you said, 30^3 is already quite fine of a grid and if needed we could just make that even finer. So in my opinion, a deterministic approach to assign a new point that is between the grid lines to an existing cluster would be more transparent and probably sufficiently precise (plus easier from the implementation I would say)."

Yes you are right ;)

**Comment by klieret** [comment]
I just opened a new issue for discussions about the new clustering algorithms just for keeping thins ordered ;)

**Comment by klieret** [comment]
Let's just use an open google drive folder to share:

https://drive.google.com/drive/folders/16EcW5lOBcPAVNOS38u3SKTT00sO-jRoB?usp=sharing 

that should allow anyone read and write access.
